components/nexus-bootstrap/src/main/java/org/sonatype/nexus/bootstrap/jetty/ConnectorManager.java

```
@@ -0,0 +1,262 @@
+    try {
+      if (HttpScheme.HTTP == httpScheme) {
+        bean(HTTP_CONFIG_ID, HttpConfiguration.class);
+        bean(HTTP_CONNECTOR_ID, ServerConnector.class);
+      }
+      else if (HttpScheme.HTTPS == httpScheme) {
+        bean(SSL_CONTEXT_FACTORY_ID, SslContextFactory.class);
+        bean(HTTPS_CONFIG_ID, HttpConfiguration.class);
+        bean(HTTPS_CONNECTOR_ID, ServerConnector.class);
+      }
+      else {
+        throw new UnsupportedHttpSchemeException(httpScheme);
+      }
+    }
+    catch (IllegalStateException e) {
+      throw new IllegalStateException("Jetty HTTPS is not enabled in Nexus", e);
+    }
+  }
+
+  private List<ConnectorConfiguration> buildDefaultConnectors() {
+    final List<ConnectorConfiguration> result = new ArrayList<>();
+    try {
+      verifyConfiguration(HttpScheme.HTTP);
+      final int port = defaultConnector(HttpScheme.HTTP).getPort();
+      result.add(new ConnectorConfiguration()
+      {
+        @Override
+        public HttpScheme getScheme() {
+          return HttpScheme.HTTP;
+        }
+
+        @Override
+        public int getPort() {
+          return port;
+        }
+
+        @Override
+        public HttpConfiguration customize(final HttpConfiguration configuration) {
+          return configuration;
+        }
+      });
+    }
+    catch (IllegalStateException e) {
+      // nop
+    }
+    try {
+      verifyConfiguration(HttpScheme.HTTPS);
+      final int port = defaultConnector(HttpScheme.HTTPS).getPort();
+      result.add(new ConnectorConfiguration()
+      {
+        @Override
+        public HttpScheme getScheme() {
+          return HttpScheme.HTTPS;
+        }
+
+        @Override
+        public int getPort() {
+          return port;
+        }
+
+        @Override
+        public HttpConfiguration customize(final HttpConfiguration configuration) {
+          return configuration;
+        }
+      });
+    }
+    catch (IllegalStateException e) {
+      // nop
```

as sonar noted above, should log this exception rather than swallowing it.